[WIP] Remove log probes from DI docs; redirect to Live Debugger (#36742)
DataDog/documentation: content/en/tracing/trace_collection/dynamic_instrumentation/_index.md

### Limitations

- Dynamic Instrumentation is not compatible with Azure App Services or serverless environments.
- Not all instrumentation types are supported in every language. See the [language-specific setup instructions](#enable-dynamic-instrumentation) for supported features and limitations.
- The Java SDK does not support Kotlin coroutines.

## Explore Dynamic Instrumentation

> Dynamic Instrumentation can help you understand what your application is doing at runtime. By adding an instrumentation, you export additional data from your application without the need to change code or redeploy it.

Reviewer: ```suggestion
Dynamic Instrumentation can help you understand what your application is doing at runtime. By adding an instrumentation at a specific code location, you can capture additional telemetry from your application without the need to change code or redeploy it.
```

### Using instrumentations

> An instrumentation allows you to collect data from specific points in your code without halting the execution of the program.

Reviewer: ```suggestion
An instrumentation allows you to collect additional telemetry from specific points in your code without halting the execution of the program.
```

> Think of using instrumentations as enhancing your observability by adding dynamic metrics and spans to a running application without needing to change code, deploy it, or restart a service. You can gather data immediately without disturbing the user experience or requiring lengthy deployments.

Reviewer: ```suggestion
Dynamic spans, span tags, and metrics are a UI-based alternative to manually adding custom instrumentation directly to your source code. The instrumentation configurations you define in the Datadog platform are received and dynamically applied to the running service - no restarts required. The new telemetry is collected when the specific part of the code defined in the instrumentation configuration is executed.

Spans, tags, and metrics generated by Dynamic Instrumentation will be captured and processed in Datadog like any other telemetry produced by your running application. The benefit of using Dynamic Instrumentation over manual custom instrumentation is that there's no need to change code, deploy it, or restart a service. You can begin capturing the additional data points within seconds, without disturbing the user experience or requiring lengthy deployments. Turning off individual dynamic instrumentations is also easy - just disable the instrumentation in the Datadog app to stop collecting that data.
```

> As a developer, you can also think of an instrumentation as a "non-breaking breakpoint". In traditional debugging, a breakpoint is a point in the program where the execution stops, allowing the developer to inspect the state of the program at that point. However, in real-world production environments, it's not practical or even possible to stop the execution of the program. Dynamic Instrumentation fills in this gap by allowing you to inspect variable state in production environments in a non-intrusive way.

Reviewer: ```suggestion
```

This paragraph is mostly relevant for logpoints, so not necessary for this doc.

### Creating an instrumentation

All instrumentation types require the same initial setup:

1. Go to the [Dynamic Instrumentation page][12].
1. Click **Create Instrumentation** in the top right, or click the three-dot menu on a service and select **Add an instrumentation for this service**.

> 1. If they are not prefilled, choose service, runtime, environment, and version.

Reviewer: ```suggestion
1. If they are not prefilled, choose service, runtime, environment, and version (optional).
```

> 1. In the source code, specify where to set the instrumentation by selecting either a class and method or a source file and line. If you opted into the [autocomplete and search Preview][17], autocomplete shows suggestions for selecting a class or method.

Reviewer: ```suggestion
1. Specify where to set the instrumentation in the source code by specifying either a class and method, or a specific line of code in a given file. In most cases, there should be an autocomplete search available for you to easily find the files, methods, or symbols you're looking for. 
 
For the best experience, we strongly recommend setting up your [Source Code Integration][7], so that you can see your code directly in Datadog and select the instrumentation code location - like you would with a breakpoint in an IDE.
```

See the individual instrumentation types below for specific creation steps for each instrumentation type.

<div class="alert alert-info">Dynamic log instrumentations are supported in <a href="/tracing/live_debugger/">Live Debugger</a>. Use Live Debugger to capture logs and variable snapshots in real time from running applications.</div>

> ### Creating metric instrumentations

Reviewer: ```suggestion
### Creating dynamic metrics
```

> A *metric instrumentation* emits a metric when it executes.

Reviewer: ```suggestion
A dynamic metric emits a metric when it executes.
```

> To create a metric instrumentation:

Reviewer: ```suggestion
To create a dynamic metric:
```

1. Select **Metric** as the instrumentation type.

> 1. Complete the [generic instrumentation setup](#creating-an-instrumentation) (choose service, environment, version, and location).

Reviewer: ```suggestion
1. Complete the [generic instrumentation setup](#creating-an-instrumentation) (choose service, environment, and location).
```

1. Specify a name for the metric, which will be prefixed with `dynamic.instrumentation.metric.probe.`.
1. Select a metric type (count, gauge, or histogram).
1. Choose the value of the metric using the [Dynamic Instrumentation expression language][15]. You can use any numeric value you'd like from the execution context, such as a method parameter, local variable, a class field, or an expression that yields a numeric value. For count metrics this is optional, and if you omit it, every invocation increments the count by one.

{{< img src="dynamic_instrumentation/metric_probe.png" alt="Creating a Dynamic Instrumentation metric instrumentation" >}}

Metric instrumentations are automatically enabled on all service instances that match the configured environment and version. Metric instrumentations are not rate limited and execute every time the method or line is invoked.

> Dynamic Instrumentation metric instrumentations support the following metric types:

Reviewer: ```suggestion
Dynamic Instrumentation supports the following metric types:
```

- **Count**: Counts how many times a given method or line is executed. Can be combined with [metric expressions][15] to use the value of a variable to increment the count.
- **Gauge**: Generates a gauge based on the last value of a variable. This metric requires a [metric expression][15].
- **Histogram**: Generates a statistical distribution of a variable. This metric requires a [metric expression][15].

> ### Creating span instrumentations

Reviewer: ```suggestion
### Creating dynamic spans
```

> A *span instrumentation* emits a span when a method is executed.

Reviewer: ```suggestion
A *dynamic span* emits a span when a method is executed.
```

> To create a span instrumentation:

Reviewer: ```suggestion
To create a dynamic span:
```

1. Select **Span** as the instrumentation type.
1. Complete the [generic instrumentation setup](#creating-an-instrumentation) (choose service, environment, version, and location).

{{< img src="dynamic_instrumentation/span_probe.png" alt="Creating a Dynamic Instrumentation span instrumentation" >}}

You can use a *span instrumentation* as an alternative to [creating new spans with Custom Instrumentation][13]. If the method throws an exception, the details of the exception are associated with the newly created span's `error` tag.

> ### Creating span tag instrumentations

Reviewer: ```suggestion
### Creating dynamic span tags
```

> A *span tag instrumentation* adds a tag value to an existing span. You can add a tag either to the _active_ span or to the _service entry_ span.

Reviewer: ```suggestion
A *dynamic span tag* adds a tag value to an existing span. You can add a tag either to the _active_ span or to the _service entry_ span.
```

Keep in mind that internal spans are not indexed by default and so might not be searchable in APM.

> To create a span tag instrumentation:

Reviewer: ```suggestion
To create a dynamic span tag:
```
